hopdrops/hopdrops
# fix: make category creation idempotent

Ensures that category creation process is idempotent (ignore the branch name).

This pull request adds functionality to create an `idempotency_key` attribute for the `CREATE_USER` event.

The purpose of the idempotency key is to prevent the Square API from creating a category with the same name (e.g., "hopdrops").

**Verification Instructions**

1. Register for an account at https://hopdrops.io
2. Repeat step 1
3. Visit https://squareup.com/dashboard/items/categories
4. Confirm that exactly *one* "hopdrops" category exists

Resolves #162
